batcher/aligned-sdk/src/sdk.rs

```diff
@@ -99,6 +111,103 @@ pub async fn submit_multiple_and_wait_verification(
+
+For batch cost estimates we estimate a batch will have ~32 proofs present.
+The `max_fee` estimates therefore are:
+    - Min: Specifies a `max_fee` equivalent to the cost of 1 proof in a 32 proof batch.
+        This estimates the lowest possible cost the user should specify for there proof with lowest priority.
+    - Default: Specifies a `max_fee` equivalent to the cost of 10 proofs in a 32 proof batch.
+        This estimates the cost the user should specify for inclusion within the batch.
+    - Instant: specifies a `max_fee` equivalent to the cost of all proofs within in a 32 proof batch.
+        This estimates the cost the user should specify to pay for the entire batch of proofs and have there proof included instantly.
+*/
+pub async fn estimate_max_fee(
+    eth_rpc_url: &str,
+    estimate: PriceEstimate,
+) -> Result<U256, errors::SubmitError> {
+    // Price of 1 proof in 32 proof batch
+    let fee_per_proof = match fee_per_proof(eth_rpc_url, 32).await {
+        Ok(price) => price,
+        Err(e) => {
+            return Err(SubmitError::GenericError(format!(
+                "Failed to compute `fee_per_proof`: {:?}",
+                e
+            )))
+        }
+    };
+
+    let proof_price = match estimate {
+        PriceEstimate::Min => fee_per_proof,
+        PriceEstimate::Default => U256::from(10) * fee_per_proof,
+        PriceEstimate::Instant => U256::from(32) * fee_per_proof,
+    };
+    Ok(proof_price)
+}
+
+/// Returns the compute `max_fee` for a proof based on the number of proofs in a batch (`num_proofs_per_batch`) and
+/// number of proofs (`num_proofs`) in that batch the user would pay for i.e (`num_proofs` / `num_proofs_per_batch`).
+// NOTE: The `max_fee` is computed from an rpc nodes max priority gas price.
+pub async fn compute_max_fee(
+    eth_rpc_url: &str,
+    num_proofs: usize,
+    num_proofs_per_batch: usize,
+) -> Result<U256, errors::SubmitError> {
+    let fee_per_proof = match fee_per_proof(eth_rpc_url, num_proofs_per_batch).await {
+        Ok(price) => price,
+        Err(e) => {
+            return Err(SubmitError::GenericError(format!(
+                "Failed to compute `fee_per_proof`: {:?}",
+                e
+            )))
+        }
+    };
+
+    Ok(fee_per_proof * num_proofs)
+}
+
+/// Retrieves the  `gas_price` from an rpc nodes. Specifically, it returns the max priority gas price.
+async fn fetch_gas_price(eth_rpc_url: &str) -> Result<U256, errors::SubmitError> {
+    let eth_rpc_provider =
+        Provider::<Http>::try_from(eth_rpc_url).map_err(|e: url::ParseError| {
+            errors::VerificationError::EthereumProviderError(e.to_string())
+        })?;
+
+    let gas_price = match eth_rpc_provider.get_gas_price().await {
+        Ok(price) => price,
+        Err(e) => {
+            return Err(SubmitError::GenericError(format!(
+                "Failed to fetch gas price: {:?}",
+                e
+            )))
+        }
+    };
+
+    Ok(gas_price)
+}
```

Comment:
I think it doesn't make much sense to define a function on our SDK just to call the rpc provider's `get_gas_price` method. Maybe a better pattern will be to instantiate a provider only once, and ask for the gas price directly to it. This data can be used directly in the other functions without the need need to instantiate a provider every time. 
@Oppen if you have any thoughts on this please feel free to share them

Reply:
Decoupling fetching the gas cost was done in response to this comment above https://github.com/yetanotherco/aligned_layer/pull/1045#pullrequestreview-2326200196 in https://github.com/yetanotherco/aligned_layer/pull/1045/commits/514ca8d0602e19c2250a60866afba232851377e2 . Happy to revert the change.

Reply:
Now passes provider